src/closure.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2013-2017, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+// TODO: support marshaller.
```

Comment:
That would indeed be very interesting and is also on my list. For dynamically using signals/action signals
Separate pull request, yes